pkg/ibmvpc/parser.go

```diff
@@ -614,12 +634,28 @@ func getNACLconfig(rc *datamodel.ResourcesContainerModel,
 }
 
 func getTgwObjects(c *datamodel.ResourcesContainerModel,
-	res map[string]*vpcmodel.VPCConfig) map[string]*TransitGateway {
+	res map[string]*vpcmodel.VPCConfig, resourceGroup string) map[string]*TransitGateway {
 	tgwMap := map[string]*TransitGateway{} // collect all tgw resources
 	for _, tgwConn := range c.TransitConnectionList {
 		tgwUID := *tgwConn.TransitGateway.Crn
 		tgwName := *tgwConn.TransitGateway.Name
 		vpcUID := *tgwConn.NetworkID
+
+		// filtering by resourceGroup
+		if resourceGroup != "" {
+			shouldSkip := true
+			for _, tgw := range c.TransitGatewayList {
+				if tgwUID == *tgw.ID {
+					if *tgw.ResourceGroup.ID == resourceGroup {
+						shouldSkip = false
+					}
```

Comment:
```suggestion
				if tgwUID == *tgw.ID && *tgw.ResourceGroup.ID == resourceGroup  {
						shouldSkip = false
					}
```

Reply:
I did split them to break in case tgwUID == *tgw.ID even if *tgw.ResourceGroup.ID != resourceGroup

Reply:
oh, ok